src/components/FormStep/FormStep.stories.mdx

```diff
@@ -0,0 +1,94 @@
+import {Meta, Canvas, Story, ArgsTable} from '@storybook/addon-docs';
+import {getWorker} from 'msw-storybook-addon';
+
+import {ConfigDecorator, RouterDecorator} from 'story-utils/decorators';
+import {getForm, getSubmission} from 'story-utils/mocks';
+
+import FormStep from '.';
+import {getSubmissionStepDetail, mockSubmissionStepGet, mockSubmissionLogicCheckPost} from './mocks';
+
+export const worker = getWorker();
+
+<Meta
+  title="Private API/FormStep"
+  component={FormStep}
+  decorators={[ConfigDecorator, RouterDecorator]}
+  argTypes={{}}
+  parameters={{
+    docs: {
+      source: {
+        type: 'dynamic',
+        excludeDecorators: true,
+      }
+    },
+  }}
+/>
+
+export const Template = (args) => {
+  const submissionStepDetailBody = getSubmissionStepDetail({
+    formioConfiguration: args.formioConfiguration});
+  worker.use(
+    mockSubmissionStepGet(submissionStepDetailBody),
+    mockSubmissionLogicCheckPost(args.submission, submissionStepDetailBody),
+  );
+  return (<FormStep
+    form={args.form}
+    submission={args.submission}
+    onLogicChecked={() => console.log('logic checked')}
+    onStepSubmitted={() => alert('step submitted')}
+    onLogout={() => alert('logged out')}
+    showDebug={args.showDebug}
+  />);
+};
+
+# FormStep component
+
+The `FormStep` component is the heart of the application.
+
+It fetches the (dynamic) configuration from the backend, reconciles it with the submission
+data and ensures that the form fields are rendered by passing the configuration and data
+down to the Formio form renderer.
+
+The component depends on a React Router parent, as the requested form step is taken from
+the URL parameters.
+
+<Canvas>
+  <Story
+    name="Default"
+    args={{
+      formioConfiguration: {
+        display: 'form',
+        components: [
+          {
+            type: 'textfield',
+            key: 'text1',
+            label: 'Simple text field',
+            description: 'A help text for the text field',
+          },
+          {
+            type: 'radio',
+            key: 'radio1',
+            label: 'Radio choices',
+            values: [
+              {value: 'option1', label: 'Option1'},
+              {value: 'option2', label: 'Option2'},
+            ]
+          }
+        ]
+      },
+      routerArgs: {
+        route: '/stap/:step',
+        initialEntries: ['/stap/step-1'],
+      },
+      form: getForm(),
+      submission: getSubmission(),
+      showDebug: false,
```

Comment:
When interacting with the component in storybook, the controls dont seem to do much if one changes the configuration of the form/the submission. Maybe they shouldn't be in the controls? The only prop that changes something seems to be the DEBUG

Reply:
`form` and `submission` are now disabled controls and I've added a workaround to get the component to re-render when the formio configuration is changed.